File: frontend/src/types/invoice.ts
export type InvoiceStatus = "draft" | "sent" | "overdue";

export interface Invoice {
  invoice_id: string;
  customer_id: string;
  customer_name: string;
  customer_email: string;
  country: string;
  customer_since: string;

  invoice_date: string;
  due_date: string;
  status: InvoiceStatus;

  total_amount: string;
  line_item_count: number;
  line_items_total: string;
  payment_count: number;
  amount_paid: string;
  last_payment_date: string | null;
  balance_due: string;
}

File: backend/app/models/invoice.py
from datetime import date
from decimal import Decimal
from typing import Literal

from pydantic import BaseModel, ConfigDict

InvoiceStatus = Literal["draft", "sent", "overdue"]


class Invoice(BaseModel):
    model_config = ConfigDict(frozen=True, extra="forbid")

    invoice_id: str
    customer_id: str
    customer_name: str
    customer_email: str
    country: str
    customer_since: date

    invoice_date: date
    due_date: date
    status: InvoiceStatus

    total_amount: Decimal
    line_item_count: int
    line_items_total: Decimal
    payment_count: int
    amount_paid: Decimal
    last_payment_date: date | None
    balance_due: Decimal
